crates/validation/tests/scenario_tests.rs

```diff
@@ -8,6 +8,126 @@ fn test_golden_all_visits() {
     insta::assert_debug_snapshot!(outcome);
 }
 
+#[test]
+fn test_dekaf_materialization_inline_config() {
+    let fixture = r##"
+test://example/catalog.yaml:
+  materializations:
+    good:
+      endpoint:
+        dekaf: &config
+          variant: foo
+          config:
+            strict_topic_names: false
+      bindings: []
+driver:
+  materializations:
+    good: &connector
+      connectorType: DEKAF
+      config: *config
+      bindings: []
+  dataPlanes:
+    "1d:1d:1d:1d:1d:1d:1d:1d":
+      default: true
+"##;
+
+    let outcome = common::run(fixture, "{}");
+    // Expect not to see any projections for the empty properties
+    insta::assert_debug_snapshot!(outcome);
+}
+
+#[test]
+fn test_dekaf_materialization_indirect_config() {
```

Comment:
tests over model shapes / parsing / config indirection should live in the `sources` crate.

`validation` crate tests focus on validation behaviors.